frontend/packages/dev-console/src/components/pipelineruns/logs/Logs.tsx

```diff
@@ -0,0 +1,104 @@
+import * as React from 'react';
+import { Base64 } from 'js-base64';
+import { resourceURL, modelFor, PodKind, ContainerSpec } from '@console/internal/module/k8s';
+import { WSFactory } from '@console/internal/module/ws-factory';
+import { LOG_SOURCE_TERMINATED } from '@console/internal/components/utils';
+import { coFetchText } from '@console/internal/co-fetch';
+import './Logs.scss';
+
+type LogsProps = {
+  resource: PodKind;
+  resourceStatus: string;
+  container: ContainerSpec;
+  render: boolean;
+  onComplete: (containerName: string) => void;
+};
+
+const Logs: React.FC<LogsProps> = ({ resource, resourceStatus, container, onComplete, render }) => {
+  const { name } = container;
+  const { kind, metadata = {} } = resource;
+  const { name: resName, namespace: resNamespace } = metadata;
+  const contentRef = React.useRef<HTMLDivElement>(null);
+  const [fetching, setFetching] = React.useState<boolean>(false);
+  const appendMessage = React.useCallback(
+    (blockContent) => {
+      const contentLines = blockContent.split('\n').filter((line) => !!line);
+      if (contentRef.current && contentLines.length >= 0) {
+        const elements = contentLines.map((content) => {
+          const customElement = document.createElement('div');
+          customElement.textContent = content;
+          return customElement;
+        });
+        elements.forEach((element) => {
+          contentRef.current.append(element);
+        });
+
+        const lastElement = elements[elements.length - 1];
+        if (render && lastElement) {
+          lastElement.scrollIntoView({ behavior: 'smooth', block: 'end' });
+        }
+      }
+    },
+    [contentRef, render],
+  );
+
+  React.useEffect(() => {
+    let ws: WSFactory;
+    let ignore: boolean = false;
+    if (!fetching) {
+      const urlOpts = {
+        ns: resNamespace,
+        name: resName,
+        path: 'log',
+        queryParams: {
+          container: name,
+          follow: 'true',
+        },
+      };
+      const watchURL = resourceURL(modelFor(kind), urlOpts);
+
+      if (resourceStatus === LOG_SOURCE_TERMINATED) {
+        coFetchText(watchURL)
+          .then((res) => {
+            if (!ignore) {
+              appendMessage(res);
+              onComplete(name);
+            }
+          })
+          .catch(() => {});
```

Comment:
Hmmm... did we discuss what we would do on error? I thought we said we'd show something because it's a 1-time shot at getting the logs -- if it doesn't it'll just remain empty. Also, I think it will block the rest of the log from loading too.

Reply:
Done(ish) will be fixed soon.